integration-tests/deployment/ccip/add_chain.go

```diff
@@ -193,3 +166,105 @@ func NewChainInboundProposal(
 		"0s", // TODO: Should be parameterized.
 	)
 }
+
+func NewSetCandidateProposal(
+	state CCIPOnChainState,
+	e deployment.Environment,
+	homeChainSel, feedChainSel, newChainSel uint64,
+	tokenConfig TokenConfig,
+	rmnHomeAddress []byte,
+) (*timelock.MCMSWithTimelockProposal, *CreateDonArgs, error) {
+	// Home chain new don.
+	// - Add new DONs for destination to home chain
+	nodes, err := deployment.NodeInfo(e.NodeIDs, e.Offchain)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	newDONArgs, err := BuildAddDONArgs(
+		e.Logger,
+		state.Chains[newChainSel].OffRamp,
+		e.Chains[newChainSel],
+		feedChainSel,
+		tokenConfig.GetTokenInfo(e.Logger, state.Chains[newChainSel].LinkToken),
+		nodes.NonBootstraps(),
+		rmnHomeAddress,
+	)
+	if err != nil {
+		return nil, nil, err
+	}
+	tabi, err := ccip_home.CCIPHomeMetaData.GetAbi()
+	if err != nil {
+		return nil, nil, err
+	}
+	commitConfig, execConfig, donId, err := FormCreateDonArgs(state.Chains[homeChainSel].CapabilityRegistry, newDONArgs)
```

Comment:
should be able to remove this and just use get the donID inline here as thats all we strictly need